ExomeDepth/igv_xml_session.py

```diff
@@ -215,34 +119,86 @@ def make_single_igv_session(args, igv_extension, vcf_extension):
+
+    #  Scale XML variables
+    min_axis, mid_axis, max_axis = settings.igv_settings[statistic]  # Get axis values out of settings.py
+
+    #  Substitue variables in IGV template
+    substitute_dic = {
+        "snv_vcf_child": f"../single_sample_vcf/{snv_vcf_child_file}",
+        "snv_vcf_father": f"../single_sample_vcf/{snv_vcf_father_file}",
+        "snv_vcf_mother": f"../single_sample_vcf/{snv_vcf_mother_file}",
+        "snv_vcf_child_id": f"SNV/MNV_child:{snv_vcf_child_file}",
+        "snv_vcf_father_id": f"SNV/MNV_father:{snv_vcf_father_file}",
+        "snv_vcf_mother_id": f"SNV/MNV_mother:{snv_vcf_mother_file}",
+        "hc_cnv_vcf_child": f"HC/{hc_cnv_vcf_child_file}",
+        "hc_cnv_vcf_father": f"HC/{hc_cnv_vcf_father_file}",
+        "hc_cnv_vcf_mother": f"HC/{hc_cnv_vcf_mother_file}",
+        "hc_cnv_vcf_child_id": f"CNV_child:{hc_cnv_vcf_child_file}",
+        "hc_cnv_vcf_father_id": f"CNV_father:{hc_cnv_vcf_father_file}",
+        "hc_cnv_vcf_mother_id": f"CNV_mother:{hc_cnv_vcf_mother_file}",
+        "hc_ratio_child": f"igv_tracks/{hc_ratio_child_file}",
+        "hc_ratio_father": f"igv_tracks/{hc_ratio_father_file}",
+        "hc_ratio_mother": f"igv_tracks/{hc_ratio_mother_file}",
+        "hc_ratio_child_track": f"igv_tracks/{hc_ratio_child_file}_{statistic}_test",
+        "hc_ratio_father_track": f"igv_tracks/{hc_ratio_father_file}_{statistic}_test",
+        "hc_ratio_mother_track": f"igv_tracks/{hc_ratio_mother_file}_{statistic}_test",
+        "hc_ratio_child_track_id": f"Probe_ratio_child:HC_{statistic}_{child}",
+        "hc_ratio_father_track_id": f"Probe_ratio_father:HC_{statistic}_{father}",
+        "hc_ratio_mother_track_id": f"Probe_ratio_mother:HC_{statistic}_{mother}",
+        "upd": f"../upd/{upd_file}",
+        "upd_track": f"../upd/{upd_file}_inheritence",
+        "upd_track_id": f"Mendelian_violations_{familyid}",
+        "baf": f"../baf/{baf_file}",
+        "baf_track_child": f"../baf/{baf_file}_baf",
+        "baf_track_child_id": f"BAF_child:{child}",
+        "bam": f"../bam_files/{bam_file_child}",
+        "bam_coverage": f"../bam_files/{bam_file_child}_coverage",
+        "bam_id": bam_file_child,
+        "fontsize": args.fontsize,
+        "mid_axis": mid_axis,
+        "max_axis": max_axis,
+        "min_axis": min_axis,
+        "session": f"{familyid}_{args.runid}_{statistic}_igv.xml"
+    }
+    new_file = template_file.substitute(substitute_dic)
+    return new_file
+
+
+def make_family_igv_session(args):
+    samples = parse_ped(args.ped_file)
+    for sample in samples:
+        if len(samples[sample]['parents']) == 2:  # Sample = child both with parents
+            familyid = samples[sample]['family']
+            child = sample
+            father = samples[sample]['parents'][0]
+            mother = samples[sample]['parents'][1]
+            if child in "".join(args.bam_files):  # Makes sure sample in within analysis, not only in PED file
```

Comment:
Do you also need to check for parents -> just to be sure?

Reply:
probably a good idea